### ISSUE TITLE
MessagePack not supported by MonoPlatform?

### ISSUE THREAD

**Issue by simonziegler** [aspnet/Blazor#1098]
I have experimented with using MessagePack (https://github.com/neuecc/MessagePack-CSharp) with Blazor to speed up serialization and deserialization over Http from server to client. I have checked that I can serialize and deserialize on the server with success, and that the byte array as it is created on the server is correctly decoded on the client (same sized array and the first and last bytes are the same).

However it looks like Mono doesn't support MessagePack for use on the Blazor client -  see screen shot of Firefox's console output below. Anybody got any views? Am I doing something wrong (I hope so), or is this just unsupported stuff?

Thanks,     Simon

![image](https://user-images.githubusercontent.com/11708435/42414646-42b81d54-8231-11e8-84a1-275b1a678183.png)

**Comment by Andrzej-W** [comment]
I'm not sure, but it looks like it is trying to dynamically create the assembly and I think this is not supported in Blazor (Mono running inside browser).

**Comment by simonziegler** [comment]
I'm out of my depth with this one. It would be really helpful to have this working, as I am experimenting with a web app that does a bunch of complex financial calcs (existing codebase) on the server, plus some database access, all taking around 40ms. Decoding a hierarchical JSON data structure on the client takes over 1 second, and that is for a fairly simple financial structure - a complex one is out of bounds. If messagepack can slice this down then I'll have a fantastic demonstrator. Is there a resolution here or am I just barking up the wrong tree? Thanks!

**Comment by SteveSandersonMS** [comment]
Two possibilities:

* The linker might be stripping out some code that MessagePack needs (e.g., if it's only reached via reflection, the linker doesn't know you're using it). Try disabling linking by putting `<BlazorLinkOnBuild>false</BlazorLinkOnBuild>` in a `<PropertyGroup>` in your `.csproj` file.
* Or maybe Mono WebAssembly is missing support for something that MessagePack relies on. If so, you could file a bug at https://github.com/mono/mono, being sure to mention it's a WebAssembly-specific issue. 

Hope that helps!

**Comment by SteveSandersonMS** [comment]
BTW MessagePack itself can also be configured to change which resolvers it uses. You might be able to configure it not to use `DynamicEnumResolver`, as that seems to be what's blowing up in your stack trace.

**Comment by simonziegler** [comment]
Thanks Steve

**Comment by simonziegler** [comment]
@SteveSandersonMS, thanks again for your suggestions. I have recreated this error at https://github.com/simonziegler/BlazorMessagePackTest and will submit to the Mono team as you suggest. If you care to give this a spin, you will want to look at the modifications that I have made to your out-of-the-box WeatherForecast class, the SampleDataController and the new page I added at the bottom of the menu. I demonstrate that:

1. MessagePack successfully performs Serialize and Deserialize calls on the server (the standard WeatherForecast HttpGet call does an otherwise pointless round trip to prove the point), and
2. MessagePack Deserialization fails on the Client, notwithstanding that I display identical byte array results originating from the server 

Thanks for the great ongoing work with building Blazor - I very much hope it sees the light of day as a fully fledged product in the not too distant future.

Simon

and client (the code makes it clear).
